examples_gbs/run_tutorial_sample.py

```diff
@@ -0,0 +1,140 @@
+# In each case, a sample is a sequence of integers of ``len(modes) = 5``. Threshold samples are
+# ``0``'s and ``1``'s, corresponding to whether or not photons were detected in a mode. A ``1``
+# here is conventionally called a "click". PNR samples are non-negative integers counting the number
+# of photons detected in each mode. For example, suppose a PNR sample is ``[2, 1, 1, 0, 0]``,
+# meaning that 2 photons were detected in mode 0, 1 photons were detected in modes 1 and 2,
+# and 0 photons were detected in modes 3 and 4. If threshold detectors were used instead,
+# the sample would be: ``[1, 1, 1, 0, 0]``.
+#
+# Sampling subgraphs
+# ------------------
+#
+# So when would threshold detection or PNR detection be preferred in GBS? Since threshold samples
+# can be post-processed from PNR samples, we might expect that PNR detection is always the
+# preferred choice. However, *simulating* PNR-based GBS is significantly slower, and it turns out
+# that threshold samples can provide enough useful information for a range of applications.
+#
+# The applications layer of Strawberry Fields focuses primarily on solving graph-based problems.
+# In this setting, we typically want to use GBS to sample subgraphs of a graph, which are likely
+# to be dense due to the probability distribution of GBS :cite:`arrazola2018using`. In this case,
+# threshold sampling is enough, since it lets us select nodes of the subgraph. Let's take more of
+# a look at this by using a small fixed graph as an example:
+
+from strawberryfields.gbs import plot
+import networkx as nx
+import plotly
+
+adj = np.array(
+    [
+        [0, 1, 0, 0, 1, 1],
+        [1, 0, 1, 0, 1, 1],
+        [0, 1, 0, 1, 1, 0],
+        [0, 0, 1, 0, 1, 0],
+        [1, 1, 1, 1, 0, 1],
+        [1, 1, 0, 0, 1, 0],
+    ]
+)
+
+graph = nx.Graph(adj)
+
+plotly.offline.plot(plot.plot_graph(graph), filename="random_graph.html")
+
+##############################################################################
+# .. raw:: html
+#     :file: ../../examples_gbs/random_graph.html
+#
+# We can sample this graph from GBS using:
+
+n_mean = 4
+samples = 20
+
+s = sample.sample(adj, n_mean, samples)
+
+print(s[:5])
+
+##############################################################################
+# Each sample in ``s`` is a list of modes with ``1``'s for nodes that have clicked and ``0``'s
+# for nodes that haven't. We want to convert a sample to another representation where the result
+# is a list of modes that have clicked. This list of modes can be used to select a subgraph of
+# the graph. For example, if ``[0, 1, 0, 1, 1, 0]`` is a sample from GBS then ``[1, 3, 4]`` are
+# the selected nodes of the corresponding subgraph.
+#
+# However, the number of clicks in GBS is a random variable and we are not always guaranteed to
+# have enough clicks in a sample for the resultant subgraph to be of interest. We can filter out
+# the uninteresting samples using:

```

Comment:
This applies to line 84 above as well --> Do we want to say "using:" or "using the `postselect` function"